go/vt/vtgate/scatter_conn.go

```diff
@@ -72,6 +72,15 @@ type shardActionFunc func(rs *srvtopo.ResolvedShard, i int) error
 // the results and errors for the caller.
 type shardActionTransactionFunc func(rs *srvtopo.ResolvedShard, i int, shardActionInfo *shardActionInfo) (*shardActionInfo, error)
 
+type (
+	resultsObserver interface {
+		observe(*sqltypes.Result)
+	}
```

Comment:
should this method not take in the shard information to track where this result set comes from?